### PR TITLE
Standard folder structure

### PR DESCRIPTION
**Pull request by tuler**
This moves src files to a standard project structure as described by the [scala gradle plugin](https://docs.gradle.org/current/userguide/scala_plugin.html#N1664C).

This was raised in issue #643 

I would still change thinks like:

- rename package "whisk" to "openwhisk" or "com.openwhisk". Whisk.com is an existing business, and there is even a organization "whisk" at docker hub.

- rename project "scala" to "core" or "common"

I'm not sure if changes are required to ansible scripts or CI/travis scripts.

### CODE REVIEW

File: core/controller/build.gradle [hunk 1]
```diff
@@ -10,23 +10,11 @@ repositories {
...
-            srcDir 'src/resources'
-        }
-    }
-}
-
 dependencies {
     compile 'org.scala-lang:scala-library:2.11.7'
     compile project(':common:scala')
+    compile project(':core:loadBalancer')
+    compile project(':core:dispatcher')
```
Comment (rabbah) [inline core/controller/build.gradle]: fairly sure this is not necessary

File: settings.gradle [hunk 2]
```diff
@@ -2,6 +2,7 @@ include 'common:scala'
 
 include 'core:controller'
 include 'core:dispatcher'
+include 'core:loadBalancer'
```
Comment (rabbah) [inline settings.gradle]: @markusthoemmes more reason to merge loadbalance source with controller?

File: common/scala/src/main/scala/whisk/core/entity/ActivationLogs.scala [hunk 3]
```diff
@@ -28,7 +28,7 @@ import spray.json.RootJsonFormat
 import spray.json.deserializationError
 import spray.json.pimpAny
 
-protected[core] case class ActivationLogs(private val logs: Vector[String] = Vector()) extends AnyVal {
+protected[core] case class ActivationLogs(val logs: Vector[String] = Vector()) extends AnyVal {
```
Comment (rabbah) [inline common/scala/src/main/scala/whisk/core/entity/ActivationLogs.scala]: private -> public orthogonal change?

File: core/controller/build.gradle (hunk 1 above)
Comment (tuler) [inline core/controller/build.gradle]: If I remove that Backend.scala and Controller.scala don't compile. They depend on LoadBalancerService.

File: common/scala/src/main/scala/whisk/core/entity/ActivationLogs.scala (hunk 3 above)
Comment (tuler) [inline common/scala/src/main/scala/whisk/core/entity/ActivationLogs.scala]: Without this I get the following warning:
sbt-api: Unhandled type class scala.reflect.internal.Types$MethodType : ($this: whisk.core.entity.ActivationLogs)Vector[String]

And this triggers an error in the dependent projects.

I found these related issues:
http://stackoverflow.com/questions/29720289/warning-about-unhandled-type-class-during-compilation
http://www.scriptscoop2.com/t/3557fd15ab2c/scala-warning-about-unhandled-type-class-during-compilation.html

Seems like a compiler bug.

File: core/controller/build.gradle (hunk 1 above)
Comment (markusthoemmes) [inline core/controller/build.gradle]: I think @rabbah is just talking about the dispatcher dependency. Does it yield the same?

**Comment by tuler** [comment]
Build broken, not sure why :-(

**Comment by markusthoemmes** [comment]
Doesn't look related to your change, restarted the build.

File: settings.gradle (hunk 2 above)
Comment (markusthoemmes) [inline settings.gradle]: Definitly! @tuler would you mind to copy the code from `loadbalancer` to the controller (they have been merged which is why there was no `build.gradle` left for it). It should then be safe to delete the whole loadbalancer folder I think.

**Comment by markusthoemmes** [comment]
@tuler FYI we have the policy of a linear git-history, meaning: no merge commits. Your pull-request reminded me that we haven't documented that, which we have now, see https://github.com/openwhisk/openwhisk/wiki/Contributing:-Git-guidelines.

You don't need to mind for this pull-request, we will just squash on merge, but for future pull-request it'll be easier for us if `rebase` is used to keep the pull-request on track instead of `merge`.

**Comment by rabbah** [comment]
PG 111.

**Comment by rabbah** [comment]
Rebased, squashed and merged as https://github.com/openwhisk/openwhisk/commit/3f50c7b990145cebbd269a5978b7badd84123c5e.

Eclipse users should `gradle cleanEclipse` to remove old projects and reinit with `gradle eclipse`.

**Comment by rabbah** [comment]
Thanks @tuler for the contribution.

**Comment by csantanapr** [comment]
I believe this is merged now with 3f50c7b990145cebbd269a5978b7badd84123c5e
Thanks @tuler for tine contrib. 👍

**Pull request closed without merging**